### PR TITLE
Update for compiler API change that removes NonNullable

### PR DESCRIPTION
**Pull request by samarsha**
See microsoft/qsharp-compiler#729.

### CODE REVIEW

**Review by tcNickolas — APPROVED** [review]
Looks good. Let me know if you want to merge or wait for Bettina's review?

**Comment by samarsha** [comment]
I think we should probably wait to merge this until after microsoft/qsharp-compiler#729 is merged, otherwise the end-to-end build might break.

**Pull request merged**
